src/coredump/coredump.c

```diff
@@ -1342,14 +1343,52 @@ static int gather_pid_metadata_from_argv(
                         t = buf;
                 }
 
+                if (i == META_ARGV_PIDFD) {
+                        if (!isempty(t)) {
+                                assert(kernel_fd < 0);
+
+                                kernel_fd = parse_fd(t);
+                                if (kernel_fd < 0)
+                                        return log_error_errno(kernel_fd,
+                                                               "Failed to parse PIDFD \"%s\": %m",
+                                                               t);
+
+                                r = pidref_set_pidfd(&context->pidref, kernel_fd);
+                                if (r < 0)
+                                        return log_error_errno(r,
+                                                               "Failed to initialize pidref from pidfd %d: %m",
+                                                               kernel_fd);
+                        }
+
+                        /* If there are containers involved with different versions of the code they might
+                         * not be using pidfds, so it would be wrong to set this boolean, skip it. */
+                        if (arg_enter_namespace)
+                                continue;
```

Comment:
uh, that's a misunderstanding. If EnterNamespace= is set we'll try to symbolize coredumps from the host, by entering the namespace of the container and finding the right data there.

But that only takes effect for containers that do not want to process the coredumps on their own, that's the logic implemented in forward_coredump_to_container(), can_forward_coredump() and so on.

hence, you are checking the wrong stuff here.

seriously, just drop the flag, it's really not worth the effort.

Reply:
Ah you meant a108c43e36d3ceb6e34efe37c014fc2cda856000 - I confused that with 68511cebe58977ea68ae4f57c6462e979efd1cff so I was looking at the wrong thing